### ISSUE TITLE
GitLab and Autodiscover

### ISSUE THREAD

**Issue by evilebottnawi** [singapore/renovate#371]
When i use selfhost GitLab and run renovate with `--autodiscover` argument, i got only first 20 repositories. 
Command:
```shell
renovate --platform gitlab --endpoint https://gitlab.my.company/api/v4 --token my_token --autodiscover
```

Using api command:
```shell
curl "https://gitlab.my.company/api/v4/projects?private_token=token"
```

Return JSON with 20 repositories too.

Seems we need use `page` argument to discover all repositories:
```shell
curl "https://gitlab.itgalaxy.company/api/v4/projects?private_token=token&page=2"
```

**Comment by rarkins** [comment]
Fixed in 8.46.1

**Comment by evilebottnawi** [comment]
@rarkins thanks for fast fixes!
